holidays/financial/national_stock_exchange_of_india.py

```diff
@@ -0,0 +1,131 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars.gregorian import (
+    JAN,
+    FEB,
+    MAR,
+    APR,
+    MAY,
+    JUN,
+    JUL,
+    AUG,
+    SEP,
+    OCT,
+    NOV,
+    DEC,
+)
+from holidays.observed_holiday_base import ObservedHolidayBase
+
+
+class NationalStockExchangeOfIndia(ObservedHolidayBase):
+    """National Stock Exchange of India (NSE) holidays.
+
+    References:
+        * <https://www.nseindia.com/trade/holiday-calendar>
+        * <https://archives.nseindia.com/content/circulars/CMTR54757.pdf> (2023)
+
+    Historical data:
+        * <https://nsearchives.nseindia.com/content/circulars/CMTR50560.pdf> (2022)
+        * <https://nsearchives.nseindia.com/content/circulars/CMTR54757.pdf> (2023)
+        * <https://nsearchives.nseindia.com/content/circulars/CMTR59722.pdf> (2024)
+        * <https://nsearchives.nseindia.com/content/circulars/CMTR65587.pdf> (2025)
```

Comment:
Reference links should be archived via Internet Archive's Wayback Machine or archive.today/.is/.ph to ensure they're accessible for future contributors as well
```suggestion
    References:
        * <https://web.archive.org/web/20250712194409/https://www.nseindia.com/resources/exchange-communication-holidays>
        * [2022](https://web.archive.org/web/20250821071611/https://nsearchives.nseindia.com/content/circulars/CMTR50560.pdf)
        * [2023](https://web.archive.org/web/20250821071635/https://nsearchives.nseindia.com/content/circulars/CMTR54757.pdf)
        * [2024](https://web.archive.org/web/20250821071650/https://nsearchives.nseindia.com/content/circulars/CMTR59722.pdf)
        * [2025](https://web.archive.org/web/20250624132016/https://nsearchives.nseindia.com/content/circulars/CMTR65587.pdf)
```